mods/translucent-windows.wh.cpp

```diff
@@ -567,98 +651,190 @@ BOOL GetColorSetting(LPCWSTR hexColor, COLORREF& outColor)
+        {
+            std::wstring ruledproc = program;
+            GetProcStrFromPath(ruledproc);
+
+            auto RuledSettings = std::make_unique<Settings>();
+            
+            if(currproc == ruledproc)
+            {
+                LPCWSTR pszStyle = Wh_GetStringSetting(L"RuledPrograms[%d].type", i);
+                if (0 == wcscmp(pszStyle, L"acrylicblur"))
+                    RuledSettings->BgType = RuledSettings->BlurBehind;
+                else if (0 == wcscmp(pszStyle, L"acrylicsystem"))
+                    RuledSettings->BgType = RuledSettings->AcrylicSystemBackdrop;
+                else if (0 == wcscmp(pszStyle, L"mica"))
+                    RuledSettings->BgType = RuledSettings->Mica;
+                else if (0 == wcscmp(pszStyle, L"mica_tabbed"))
+                    RuledSettings->BgType = RuledSettings->MicaAlt;
+                else 
+                    RuledSettings->BgType = RuledSettings->Default;
+                
+                Wh_FreeStringSetting(pszStyle);
+                
+                RuledSettings->ExtendFrame = Wh_GetIntSetting(L"RuledPrograms[%d].ExtendFrame", i);
+                if(RuledSettings->ExtendFrame)
+                    DwmExpandFrameIntoClientAreaHook();
+                
+                RuledSettings->TitlebarFlag = Wh_GetIntSetting(L"RuledPrograms[%d].TitlebarColor.ColorTitlebar", i);
+                if(RuledSettings->TitlebarFlag)
+                {
+                    LPCWSTR pszTitlebarStyle = Wh_GetStringSetting(L"RuledPrograms[%d].TitlebarColor.titlerbarstyles", i);
+                    RuledSettings->TitlebarFlag = GetColorSetting(pszTitlebarStyle, RuledSettings->TitlebarColor);
+                    Wh_FreeStringSetting(pszTitlebarStyle);
+                }
+
+                RuledSettings->CaptionTextFlag = Wh_GetIntSetting(L"RuledPrograms[%d].TitlebarTextColor.ColorTitlebarText", i);
+                if(RuledSettings->CaptionTextFlag)
+                {
+                    LPCWSTR pszTitlebarTextColorStyle = Wh_GetStringSetting(L"RuledPrograms[%d].TitlebarTextColor.titlerbarcolorstyles", i);
+                    RuledSettings->CaptionTextFlag = GetColorSetting(pszTitlebarTextColorStyle, RuledSettings->CaptionTextColor);
+                    Wh_FreeStringSetting(pszTitlebarTextColorStyle);
+                }
+
+                RuledSettings->BorderFlag = Wh_GetIntSetting(L"RuledPrograms[%d].BorderColor.ColorBorder", i);
+                if(RuledSettings->BorderFlag)
+                {
+                    LPCWSTR pszBorderStyle = Wh_GetStringSetting(L"RuledPrograms[%d].BorderColor.borderstyles", i);
+                    RuledSettings->BorderFlag = GetColorSetting(pszBorderStyle, RuledSettings->BorderColor);
+                    Wh_FreeStringSetting(pszBorderStyle);
+                }
+
+                RuledSettings->MenuBorderFlag = RuledSettings->BorderFlag;
+
+                SettingsMap.emplace(currProcID, std::move(RuledSettings));

```

Comment:
Why have a map if the only key that's used for it is the current process id?
Why not just have a single global `Settings` struct for the current process?
The current approach works too, I can merge it as is, it's just a suggestion. Let me know if you'd like to change it.